pam/go-exec/module.c

```diff
@@ -438,21 +439,9 @@ wait_child_thread (gpointer data)
         }
     }
 
-  if (thread_data->connection_ptr)
-    {
-      g_autoptr(GDBusConnection) connection = NULL;
-
-#if GLIB_CHECK_VERSION (2, 74, 0)
-      connection = g_atomic_pointer_exchange (thread_data->connection_ptr, NULL);
-#else
-      /* TODO: This is to support CI and old LTS (24.04) */
-      connection = g_atomic_pointer_get (thread_data->connection_ptr);
-      g_atomic_pointer_set (thread_data->connection_ptr, NULL);
-#endif
```

Comment:
The commit message doesn't explain this change

Reply:
Well it was:

    At the same time here there's no risk that we're accessing to a disposed
    connection in the thread, because at this point the wait-child thread is
    running while the action-thread is waiting for it, before disposing the
    connection.

I can try make it clearer though